playwright/tests/e2e/steps/native-token.steps.ts

```diff
@@ -142,45 +152,53 @@ Given(/^the user open the detail page of a token in the native tokens site$/, as
   await nativeTokenPage(page).goToTokensDetailPageFromTableByAssetName();
 });
 When(/^the user selects the Block number in the transactions table$/, async ({ page }) => {
+  const nativeDetailTrxTableValueEpcch = page.getByTestId("token.transaction.block#0");
+  blockNo = await nativeDetailTrxTableValueEpcch.getAttribute("aria-label");
   await nativeTokenPage(page).goToBlockDetailFromTransactionTableByBlock();
 });
 Then(/^the user should be redirected to the Block details page of the selected block number$/, async ({ page }) => {
-  await nativeTokenPage(page).getDetailBlockPageTitle();
+  await blockDetailPage(page).checkBlockDetail({ blockNo });
 });
 
 Given(/^the user open the detail page of a token in the native tokens web$/, async ({ page }) => {
   await nativeTokenPage(page).goToNativeTokens();
   await nativeTokenPage(page).goToTokensDetailPageFromTableByAssetName();
 });
 When(/^the user selects the Epoch number in the transactions table$/, async ({ page }) => {
+  const nativeDetailTrxTableValueEpoch = page.getByTestId("token.transaction.epoch#0");
```

Comment:
This interaction should be inside the native tokens page and be used from here like a function, the steps should no use the web elements only the native token page class should use them